americangut/question/ag_question.py

```diff
@@ -0,0 +1,117 @@
+    true_values = {'yes', 'y', 'true', 1, 1.0, 'Y', 'YES', 'Yes', True}
+    false_values = {'no', 'n', 'false', 0, 0.0, 'N', 'NO', 'No', False}
+
+    def __init__(self, name, description, dtype, clean_name=None, **kwargs):
+        """A base object for describing single question outputs
+
+        Parameters
+        ----------
+        name : str
+            The name of column where the group is stored
+        description : str
+            A brief description of the data contained in the question
+        dtype : object
+            The datatype in which the responses should be represented. (i.e.
+            float, int, str).
+        clean_name : str, optional
+            A nicer version of the way the column should be named.
+        remap : function
+            A function used to remap the data in the question.
+        free_response: bool, optional
+            Whether the question is a free response question or controlled
+            vocabulary
+        mimmarks : bool, optional
+            If the question was a mimmarks standard field
+        ontology : str, optional
+            The type of ontology, if any, which was used in the field value.
+
+        """
+
+        # Checks the arguments
+        if not isinstance(name, str):
+            raise TypeError('name must be a string.')
+        if not isinstance(description, str):
+            raise TypeError('description must be a string')
+        if not inspect.isclass(dtype):
+            raise TypeError('dtype must be a class')
+        if not isinstance(clean_name, str) and clean_name is not None:
+            raise TypeError('If supplied, clean_name must be a string')
+
+        # Handles the main information about the data
+        self.name = name
+        self.description = description
+        self.dtype = dtype
+
+        self.type = 'Question'
+        if clean_name is None:
+            self.clean_name = name.replace('_', ' ').title()
+        else:
+            self.clean_name = clean_name
+
+        # Sets up
+        self.free_response = kwargs.get('free_response', False)
+        self.mimmarks = kwargs.get('mimmarks', False)
+        self.ontology = kwargs.get('ontology', None)
+        self.remap_ = kwargs.get('remap', None)
+
+    def check_map(self, map_):
+        """Checks the group exists in the metadata
+
+        Parameters
+        ----------
+        map_ : DataFrame
+            A pandas object containing the data to be analyzed. The
+            Question `name` should be a column in the `map_`.
+
+        """
+        if self.name not in map_.columns:
+            raise ValueError('%s is not a column in the supplied map!'
+                             % self.name)
+
+    def remap_data_type(self, map_, watch=True):
+        """Makes sure the target column in map_ has the correct datatype
+
+        map_ : DataFrame
+            A pandas dataframe containing the column described by the question
+            name.
+        watch : bool
```

Comment:
The functionality gets leveraged in the Categorical subclass.
And example comes up in the function, [remove ambiguity](https://github.com/biocore/American-Gut/pull/192/files#diff-062a13454b23b479686ccaff0747068eR80). Here, you want to cast the data to the correct datatype, and then remove ambiguous entries. (In case, maybe, ambiguous entries are coded as -1, or something). I may also be over casting to the data type.